src/Microsoft.Diagnostics.Runtime/src/Common/HeapBase.cs

```diff
@@ -257,6 +259,34 @@ protected internal override IEnumerable<ClrObject> EnumerateObjectReferences(ulo
             return result;
         }
 
+        protected internal override IEnumerable<ClrObjectReference> EnumerateObjectReferencesWithFields(ulong obj, ClrType type, bool carefully)
+        {
+            if (type == null)
+                type = GetObjectType(obj);
+            else
+                Debug.Assert(type == GetObjectType(obj));
+
+            if (!type.ContainsPointers)
+                return s_emptyObjectReferenceSet;
+
+            GCDesc gcdesc = type.GCDesc;
+            if (gcdesc == null)
+                return s_emptyObjectReferenceSet;
+
+            ulong size = type.GetSize(obj);
+            if (carefully)
+            {
+                ClrSegment seg = GetSegmentByAddress(obj);
+                if (seg == null || obj + size > seg.End || !seg.IsLarge && size > 85000)
```

Comment:
Nitpick:  I prefer to put the unnecessary parenthesis in boolean expressions so that it's always obvious of the intent:

`if (seg == null || obj + size > seg.End || (!seg.IsLarge && size > 85000))`